sdk/storage/storage-queue/src/QueueServiceClient.ts

```diff
@@ -324,4 +307,65 @@ export class QueueServiceClient extends StorageClient {
       ...options
     });
   }
+
+  async *listSegments(
+    marker: string | undefined = undefined,
+    options: ServiceListQueuesSegmentOptions = {}
+  ): AsyncIterableIterator<Models.ServiceListQueuesSegmentResponse> {
+    let listQueuesResponse;
+    do {
+      listQueuesResponse = await this.listQueuesSegment(marker, options);
+      marker = listQueuesResponse.nextMarker;
+      yield await listQueuesResponse;
+    } while (marker);
+  }
+
+  async *listItems(
+    options: ServiceListQueuesSegmentOptions = {}
+  ): AsyncIterableIterator<Models.QueueItem> {
+    let marker: string | undefined;
+    for await (const segment of this.listSegments(marker, options)) {
+      yield* segment.queueItems;
+    }
+  }
+
+  /**
+   * Returns an async iterable iterator to list all the queues
+   * under the specified account.
+   *
+   * .byPage() returns an async iterable iterator to list the queues in pages.
+   *
```

Comment:
Can you added simple samples @sample in JSDoc ? ".byPage()` description seems not clear enough